### ISSUE TITLE
shap multi class xgboost classifier issue

### ISSUE THREAD

**Issue by ssetty** [shap/shap#2071]
Hello,

I am unable to generate shap force plot for multi class xgboost classifier. Data has numerical and categorical factors. Kindly send correct code for  explainability of a single model prediction. Force plot can be used for error analysis, finding the explanation to specific instance prediction.

Below is sample code.
import pandas as pd 
import numpy as np 

import sklearn 
from sklearn.model_selection  import train_test_split 
from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import LabelEncoder, OrdinalEncoder

import xgboost as xgb 
import xgboost 

import sklearn.ensemble
import sklearn as skl

import shap
import time 
import os 
import matplotlib.pyplot as plt
import seaborn as sns 
 
le = LabelEncoder()
df['target_col'] = le.fit_transform(df['target_col'])
print(le.classes_)

y_sel = df['target_col'] 
categorical_cols = x_sel.select_dtypes(include=['object']).columns.tolist()
numerical_cols = x_sel.select_dtypes(exclude=['object']).columns.tolist()

categorical_nan_value = 'NotAvailable'
numerical_nan_value = -1
    
oes = {}
for cat in categorical_cols:
    oes[cat] = OrdinalEncoder(handle_unknown="use_encoded_value", unknown_value=-1)
    x_sel[cat] = x_sel[cat].fillna(categorical_nan_value)
    x_sel[cat] = oes[cat].fit_transform(x_sel[cat].values.reshape(-1, 1))

from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
x_sel[numerical_cols] = scaler.fit_transform(x_sel[numerical_cols])

x_sel_tr,x_sel_te,y_sel_tr,y_sel_te = train_test_split(x_sel,y_sel,test_size =.33,random_state=1)

xgb_model = xgboost.XGBClassifier( objective='multi:softprob', num_class=8, n_estimators=1000)
xgb_model.fit(x_sel_tr, y_sel_tr)

predictions = xgb_model.predict(x_sel_te)
predictions_proba=xgb_model.predict_proba(x_sel_te)
accuracy=skl.metrics.accuracy_score(y_sel_te, predictions)
print(accuracy*100)

explainerXGB = shap.TreeExplainer(xgb_model)
shap_values_XGB_test = explainerXGB.shap_values(x_sel_te)
shap_values_XGB_train = explainerXGB.shap_values(x_sel_tr)
 
print(type(shap_values_XGB_test)) # output: <class 'list'>
print(type(shap_values_XGB_test[0])) # output: <class 'numpy.ndarray'>
print(len(shap_values_XGB_test)) # output: 3

shap.initjs()
i = 18
shap.force_plot(explainerXGB.expected_value, shap_values_XGB_train[i], x_sel_te[i], feature_names = x_sel_te.cols)

Regards

**Comment by c56pony** [comment]
To `force_plot` a multiclass classification, you need to specify which class you want to plot first. For example, if you want to plot for `class_id = 0`, you can do the following.
```
class_id = 0
shap.force_plot(explainerXGB.expected_value[class_id], shap_values_XGB_test[class_id], x_sel_te)
```
In addition, to plot for a specific individual prediction, you can do the following.
```
class_id = 0
i = 18
shap.force_plot(explainerXGB.expected_value[class_id], shap_values_XGB_test[class_id][i], x_sel_te.iloc[i])
```
